pom.xml

```diff
@@ -104,6 +104,11 @@
             <groupId>org.hamcrest</groupId>
             <artifactId>hamcrest-core</artifactId>
           </exclusion>
+          <exclusion>
+            <!-- provided by xstream in Jenkins core -->
```

Comment:
I may be misinterpreting the message or not taking the correct action for the message.  Without that change, the maven job fails with:

```
[INFO] Ignoring requireUpperBoundDeps in javax.servlet:servlet-api
[WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for javax.xml.bind:jaxb-api:2.2.12 [provided] paths to dependency are:
+-org.jenkinsci.plugins:pipeline-model-definition:1.9.4-SNAPSHOT
  +-com.jcabi:jcabi-matchers:1.5.3 [test]
    +-javax.xml.bind:jaxb-api:2.2.12 [provided]
and
+-org.jenkinsci.plugins:pipeline-model-definition:1.9.4-SNAPSHOT
  +-org.jenkins-ci.main:jenkins-core:2.321 [provided]
    +-com.thoughtworks.xstream:xstream:1.4.18 [provided] (managed) <-- com.thoughtworks.xstream:xstream:1.4.18 [provided]
      +-javax.xml.bind:jaxb-api:2.3.1 [provided]
]
```

I thought that meant that jaxb-api is included in xstream 1.4.18 and that is provided inside jenkins-core.  What have I misunderstood here?

Reply:
dependency:tree / enforcer are playing games with you... 

https://github.com/x-stream/xstream/blob/XSTREAM_1_4_18/xstream/pom.xml#L113-L117  has a dependency on it but scope provided - which is iffy and as it is provided would not be bundled in jenkins core.  it is looking like in the enforcer the jaxb dependency is pulled in by core (which is provided) but as it is provided it is not (but everything under core would be provided if it was compile or provided!)

I think the action you did is correct at this point in time (but not the comment), as we only need it for tests - but this is likely to blow up in the PCT at some point in the future in the BOM should it be tested against a core where xstream no longer pulls that in (it is gone in the next release of xstream). 

/cc @batmat who did the jaxb split (I think the dependency should be excluded in the core  based on [the comment](https://github.com/x-stream/xstream/blob/XSTREAM_1_4_18/xstream/pom.xml#L116) in the pom - it has gone in newer versions) and @basil who is spending some time looking at the jakarta.ee work as this may become a larger issue.